public/apps/login/login-page.tsx

```diff
@@ -89,60 +104,162 @@ export function LoginPage(props: LoginPageDeps) {
+          href={loginEndPoint}
+          iconType={buttonConfig.showbrandimage ? buttonConfig.brandimage : ''}
+        >
+          {buttonConfig.buttonname}
+        </EuiButton>
+      </EuiFormRow>
+    );
+  };
+
+  const formOptions = (options: string | string[]) => {
+    let formBody = [];
+    const formBodyOp = [];
+    let authOpts = [];
+
+    if (typeof options === 'string') {
+      if (options === '') {
+        authOpts.push(AuthType.BASIC);
+      } else {
+        authOpts.push(options.toLowerCase());
+      }
+    } else {
+      if (options && options.length === 1 && options[0] === '') {
+        authOpts.push(AuthType.BASIC);
+      } else {
+        authOpts = [...options];
+      }
+    }
+
+    for (let i = 0; i < authOpts.length; i++) {
+      switch (authOpts[i].toLowerCase()) {
+        case AuthType.BASIC: {
+          formBody.push(
+            <EuiFormRow>
+              <EuiFieldText
+                data-test-subj="user-name"
+                aria-label="username_input"
+                placeholder="Username"
+                prepend={<EuiIcon type="user" />}
+                onChange={(e) => setUsername(e.target.value)}
+                value={username}
+                isInvalid={usernameValidationFailed}
+              />
+            </EuiFormRow>
+          );
+          formBody.push(
+            <EuiFormRow isInvalid={passwordValidationFailed}>
+              <EuiFieldText
+                data-test-subj="password"
+                aria-label="password_input"
+                placeholder="Password"
+                prepend={<EuiIcon type="lock" />}
+                type="password"
+                onChange={(e) => setPassword(e.target.value)}
+                value={password}
+                isInvalid={usernameValidationFailed}
+              />
+            </EuiFormRow>
+          );
+          const buttonId = `${AuthType.BASIC}_login_button`;
+          formBody.push(
+            <EuiFormRow>
+              <EuiButton
+                data-test-subj="submit"
+                aria-label={buttonId}
+                fill
+                size="s"
+                type="submit"
+                className={props.config.ui.basicauth.login.buttonstyle || 'btn-login'}
+                onClick={handleSubmit}
+              >
+                Log in
+              </EuiButton>
+            </EuiFormRow>
+          );
+
+          if (authOpts.length > 1) {
+            if (props.config.auth.anonymous_auth_enabled) {
+              const anonymousConfig = props.config.ui[AuthType.ANONYMOUS].login;
+              formBody.push(
+                renderLoginButton(AuthType.ANONYMOUS, ANONYMOUS_AUTH_LOGIN, anonymousConfig)
+              );
+            }
+
+            formBody.push(<EuiSpacer size="xs" />);
+            formBody.push(<EuiHorizontalRule size="full" margin="xl" />);
+            formBody.push(<EuiSpacer size="xs" />);
+          }
+          break;
```

Comment:
I might have missed something here. Does this break the for loop? Shouldn't it push compoments for multiple auth types if `authOpts.length > 1`?

Reply:
The break for each switch case will only break the switch loop, the for loop will continue.